docs/architecture.md

```diff
@@ -1,188 +1,260 @@
 
-`kubewarden-controller` notices the default `PolicyServer` resource and, as a result of that,
+<!--TODO:For the next paragraph. What type of architecture diagram is this? In the
+diagram, which is `policy-server`. The aquamarine box with 'Kubewarden Policy
+Server'? Is 'Deployment' capitalization correct? What does 'notices' mean
+exactly? For the diagram, what's the difference between solid box, dashed box,
+white box, aquamarine box? What are the arrows showing, should they be
+labelled?-->
+
+When the `kubewarden-controller` notices the default `PolicyServer` resource,
 it creates a Deployment of the `policy-server` component.
 
-As stated above, Kubewarden works as a Kubernetes Admission Webhook. Kubernetes
-dictates that all the Webhook endpoints must be secured with TLS.
-`kubewarden-controller` takes care of setting up this secure communication
-by doing these steps:
+Kubewarden works as a Kubernetes Admission Webhook.
+Kubernetes specifies using TLS to secure all Webhook endpoints.
+The `kubewarden-controller` sets up this secure communication
+by:
+
+1. Generating a self-signed Certificate Authority
+1. Use this CA to generate a TLS certificate key for the `policy-server` Service.
+
+<!--The capitalization on Secret is ok? Maybe in ``?-->
 
-1. Generate a self-signed Certificate Authority
-1. Use this CA to generate a TLS certificate and a TLS key for the
-   `policy-server` Service.
+These objects are all stored as Secret resources in Kubernetes.
 
-All these objects are stored into Kubernetes as Secret resources.
+<!--TODO:Next paragraph. `policy-server` is in the aquamarine box, right? I
+don't think the K8s ClusterIP Service is shown in the diagram anywhere right?
+Should it be?-->
 
-Finally, `kubewarden-controller` will create the `policy-server`
-Deployment and a Kubernetes ClusterIP Service to expose it inside of
-the cluster network.
+Finally, `kubewarden-controller` creates the `policy-server` Deployment
+and a Kubernetes ClusterIP Service
+to expose it inside the cluster network.
 
 ### Defining the first policy
 
-This chart shows what happens when the first policy bounded to the default `policy-server` is defined inside of the
-cluster:
+<!--TODO:First mention of 'bound'. I'd like to define what binding or to
+be bound is, how it happens, where, etc?-->
+
+This diagram shows what happens when defining the first policy
+bound to the default `policy-server` in the cluster:
 
 ![Defining the first ClusterAdmissionPolicy resource](/img/architecture_sequence_02.png)
 
-`kubewarden-controller` notices the new `ClusterAdmissionPolicy` resource and,
-as a result of that, it finds the bounded `PolicyServer` and reconciles it.
+<!--TODO:There is ClusterAdmissionPolicy inside and outside the API
+Server box.  What is this signifying? And for PolicyServer? Definition
+of reconcile? Just to find the PolicyServer that matches the
+ClusterAdmissionPolicy?-->
```

Comment:
> There is ClusterAdmissionPolicy inside and outside the API Server box.  What is this signifying? And for PolicyServer? 

I believe this is trying to represent is the act of the user sending this `PolicyServer` and `ClusterAdmissionPolicy` definition to the Kubernetes API to be processed by the controller

> Definition of reconcile? Just to find the PolicyServer that matches the ClusterAdmissionPolicy?

Reconciliation is what  the controller does. It watches for the policy server and policies definition and translate that to Kubernetes primitives that matches what the users want when they deploy these resources.

Reply:
Thanks.